301-expressroute-circuit-vnet-connection/azureDeploy.json

```diff
@@ -0,0 +1,175 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "String",
+            "metadata": {
+                "description": "Region where the resources will be deployed"
+            }
+        },
+        "gatewayType": {
+            "type": "String",
+            "defaultValue": "ExpressRoute",
+            "allowedValues": [
+                "Vpn",
+                "ExpressRoute"
+            ],
+            "metadata": {
+                "description": "The type of gateway to deploy. For connecting to ExpressRoute circuits, the gateway must be of type ExpressRoute."
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the virtual network to create."
+            }
+        },
+        "addressPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "The address space in CIDR notation for the new virtual network."
+            }
+        },
+        "subnet1Name": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the first subnet in the new virtual network."
+            }
+        },
+        "gatewaySubnet": {
+            "type": "string",
+            "defaultValue": "GatewaySubnet",
```

Comment:
Since there is only one allowed value, this can become a variable or even just hardcode in the resource properties section if any change from the name "GatewaySubnet" is going to fail the template